frontend/src/features/public-form/components/FormFields/FormFields.tsx

```diff
@@ -97,6 +99,18 @@ export const FormFields = ({
     }, {})
   }, [augmentedFormFields, fieldPrefillMap])
 
+  // payment prefills - only for variable payments
+  if (searchParams.has(PAYMENT_VARIABLE_INPUT_AMOUNT_FIELD_ID)) {
+    const paymentParamValue = Number.parseInt(
+      searchParams.get(PAYMENT_VARIABLE_INPUT_AMOUNT_FIELD_ID) ?? '',
+      10,
+    )
+    if (Number.isInteger(paymentParamValue)) {
```

Comment:
should we also add a check that the number is negative?

Reply:
Right. Makes sense. Will do

Reply:
709646799